src/io/image_io.cc

```diff
@@ -217,17 +217,27 @@ void Imread(const nnvm::NodeAttrs& attrs,
             std::vector<NDArray>* outputs) {
 #if MXNET_USE_OPENCV
   const auto& param = nnvm::get<ImreadParam>(attrs.parsed);
-
-  std::ifstream file(param.filename, std::ios::binary | std::ios::ate);
-  // if file is not open we get bad alloc after tellg
-  CHECK(file.is_open()) << "Imread: '" << param.filename
-      << "' couldn't open file: " << strerror(errno);
-  size_t fsize = file.tellg();
-  file.seekg(0, std::ios::beg);
+  std::unique_ptr<dmlc::Stream> fi(dmlc::Stream::Create(param.filename.c_str(), "r", true));
+  CHECK(fi.get() != NULL) << "Imread: '" << param.filename
+    << "' couldn't open file: " << strerror(errno);
+  std::vector<std::unique_ptr<char> > buff_vec;
+  std::vector<int> buff_size;
+  int single_buff_size = 1000000;
+  size_t size, fsize = 0;
+  char* single_buff = new char[single_buff_size];
```

Comment:
please use unique_ptr. When you pass it to the vector of unique pointers, you can call release() if necessary, or just push it (assuming the stl hasn't blocked transfer of ownership this way)